dsgrid/utils/spark.py

```diff
@@ -342,6 +342,33 @@ def models_to_dataframe(models, table_name=None):
     return df
 
 
+def get_type_from_union(python_type) -> Type:
+    """Return the Python type from a Union.
+
+    Only works if it is Union of NoneType and something.
+
+    Raises
+    ------
+    NotImplementedError
+        Raised if the code does know how to determine the type.
+    """
+    args = get_args(python_type)
+    if issubclass(args[0], enum.Enum):
+        python_type = type(next(iter(args[0])).value)
+    else:
+        types = [x for x in args if not issubclass(x, type(None))]
+        if not types:
+            msg = f"Unhandled Union type: {python_type=} {args=}"
+            raise NotImplementedError(msg)
+        elif len(types) > 1:
+            msg = f"Unhandled Union type: {types=}"
+            raise NotImplementedError(msg)
+        else:
+            python_type = types[0]
+
+    return python_type
```

Comment:
Are you trying to find the type that this field is when it's not None, so you can set the type of the Spark dataframe column correctly?

Reply:
yes